modules/dns/dns_api.rb

```diff
@@ -27,6 +27,8 @@ class Api < ::Sinatra::Base
         when 'AAAA'
           ip = IPAddr.new(value, Socket::AF_INET6).to_s
           server.create_aaaa_record(fqdn, ip)
+        when 'CNAME'
+          server.create_cname_record(fqdn, value)
```

Comment:
We'll need a default implementation for create_cname_record (probably returning an error "This provider doesn't support creation of CNAME records" or something similar), as it's going to take some time for other providers to catch up...